macros/edr/system/system_utils/buckets_cte.sql

```diff
@@ -1,75 +1,90 @@
 {% macro complete_buckets_cte(time_bucket) %}
-    {{ adapter.dispatch('complete_buckets_cte','elementary')(time_bucket) }}
+    {%- set bucket_end_expr = elementary.edr_timeadd(time_bucket.period, time_bucket.count, 'edr_bucket_start') %}
+    {%- set min_bucket_start_expr = elementary.edr_date_trunc(time_bucket.period, elementary.edr_cast_as_timestamp(elementary.edr_quote(elementary.get_min_bucket_start()))) %}
+    {%- set max_bucket_end_expr = elementary.edr_cast_as_timestamp(elementary.edr_quote(elementary.get_max_bucket_end())) %}
+    {{ adapter.dispatch('complete_buckets_cte','elementary')(time_bucket, bucket_end_expr, min_bucket_start_expr, max_bucket_end_expr) }}
 {% endmacro %}
 
-{% macro spark__complete_buckets_cte(time_bucket) %}
-    {% set edr_bucket_end_expr = elementary.edr_timeadd(time_bucket.period, time_bucket.count, 'edr_bucket_start') %}
+{% macro spark__complete_buckets_cte(time_bucket, bucket_end_expr, min_bucket_start_expr, max_bucket_end_expr) %}
```

Comment:
Cool, I didn't know we're allowed to have a different signature for the DWH__function functions in the adapter pattern